compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -461,19 +466,19 @@ impl<'a> Linker for GccLinker<'a> {
     }
     fn link_rlib(&mut self, lib: &Path) {
         self.hint_static();
-        self.cmd.arg(lib);
+        self.cmd.arg(self.sess.opts.file_path_mapping().map_prefix(lib).0.as_ref());
```

Comment:
If the remap is anything other than turning an absolute path into a relative path that still resolves to the same absolute path given the working directory of rustc this will break linking as the linker actually needs to read the file at this path. So for example remapping `/path/to/rust_source` to `/rustc/$hash` as is done by rust's CI will be broken.

This is the reason for the CI failure.